spec/services/email_normalizer_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe EmailNormalizer do
+  subject(:normalizer) { EmailNormalizer.new(email) }
+
+  describe '#normalized_email' do
+    subject(:normalized_email) { normalizer.normalized_email }
+
+    context 'with a non-gmail domain' do
+      let(:email) { 'foobar+123@example.com' }
+
+      it 'is the same email' do
+        expect(normalized_email).to eq(email)
+      end
+    end
+
+    context 'with a gmail domain' do
+      let(:email) { 'foo.bar+123@gmail.com' }
+
+      it 'removes . and anything after the +' do
+        expect(normalized_email).to eq('foobar@gmail.com')
+      end
+    end
+
+    context 'with a Google Apps domain' do
+      let(:email) { 'foo.bar.baz+123@example.com' }
+
+      before do
+        allow(normalizer).to receive(:mx_records).with('example.com').
+          and_return(%w[abcd.mail.google.com])
```

Comment:
Stubbing this definitely leaves us blind to changes in the `resolv` library... but it's part of ruby's STDLIB and I don't expect it's API to change anytime soon

If folks feel strongly, I could build a double'd version of the `DNS` class but that seemed like a lot of work